### PR TITLE
fix conversion from ndarrays to dataframes

### LINKED ISSUE #810: DynamicTable.to_dataframe cannot handle new DynamicTableRegion objects
## 1) Bug

Running `to_dataframe` on a `DynamicTable` that contains a `DynamicTableRegion` no longer works due to the changes to `DynamicTableRegion`. This is a common practice on the Units table, which uses `DynamicTableRegion` to store electrodes (as illustrated below) so I expect many users will be affected by this bug.

### Steps to Reproduce
```python
from pynwb import NWBFile

from datetime import datetime
from dateutil.tz import tzlocal

nwbfile = NWBFile('aa', 'aa', datetime(2017, 4, 15, 12, tzinfo=tzlocal()))

device = nwbfile.create_device(name='trodes_rig123')
electrode_name = 'tetrode1'
description = "an example tetrode"
location = "somewhere in the hippocampus"
electrode_group = nwbfile.create_electrode_group(electrode_name,
                                                 description=description,
                                                 location=location,
                                                 device=device)
for idx in [1, 2, 3, 4]:
    nwbfile.add_electrode(idx,
                          x=1.0, y=2.0, z=3.0,
                          imp=float(-idx),
                          location='CA1', filtering='none',
                          group=electrode_group)

nwbfile.add_unit(id=1, electrodes=[1])
nwbfile.add_unit(id=2, electrodes=[1])

nwbfile.units.to_dataframe()
```

```python
Traceback (most recent call last):
  File "/Users/bendichter/dev/pynwb/test_units_to_dataframe.py", line 26, in <module>
    nwbfile.units.to_dataframe()
  File "/Users/bendichter/dev/pynwb/src/pynwb/core.py", line 1243, in to_dataframe
    data[name] = col[:]
  File "/Users/bendichter/dev/pynwb/src/pynwb/core.py", line 404, in __getitem__
    ret.append(self.__getitem_helper(i))
  File "/Users/bendichter/dev/pynwb/src/pynwb/core.py", line 397, in __getitem_helper
    return self.target[start:end]
  File "/Users/bendichter/dev/pynwb/src/pynwb/core.py", line 1360, in __getitem__
    raise ValueError("unrecognized argument: '%s'" % key)
ValueError: unrecognized argument: 'slice(0, 1, None)'
```

## Checklist

- [x] Have you ensured the feature or change was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues/) ?
- [x] Have you included a brief and descriptive title?
- [x] Have you included a clear description of the problem you are trying to solve?
- [x] Have you included a minimal code snippet that reproduces the issue you are encountering?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?

### PR DESCRIPTION
**Pull request by bendichter**
*convert nd arrays to list of arrays when converting to df
*add test

## Motivation

fix #810 

## How to test the behavior?
```python
from pynwb.core import DynamicTable, VectorData
import numpy as np 
import pandas as pd

col = VectorData(name='name', description='desc', data=np.array([[1,1,1],[2,2,2],[3,3,3]])) 
DynamicTable('test', 'desc', np.arange(3, dtype='int'), (col, )).to_dataframe()
```

## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

### CODE REVIEW

File: src/pynwb/core.py [hunk 1]
```diff
@@ -1240,7 +1240,10 @@ def to_dataframe(self):
         data = {}
         for name in self.colnames:
             col = self.__df_cols[self.__colids[name]]
-            data[name] = col[:]
+            if isinstance(col.data, np.ndarray) and col.data.ndim > 1:
+                data[name] = [x for x in col[:]]
```
Comment (oruebel) [inline src/pynwb/core.py:1244]: I think this is the same as ``data[name] = col.tolist()`` correct? If so, I think using ``tolist`` is probably prefered

https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.ndarray.tolist.html

File: tests/unit/pynwb_tests/test_core.py [hunk 2]
```diff
@@ -207,6 +208,10 @@ def test_indexed_dynamic_table_region(self):
         dynamic_table_region = DynamicTableRegion('dtr', [0, 1], 'desc', table=table)
         assert dynamic_table_region[slice(0, 1)]
 
+    def test_nd_array_to_df(self):
+        col = VectorData(name='name', description='desc', data=np.array([[1, 1, 1], [2, 2, 2], [3, 3, 3]]))
+        DynamicTable('test', 'desc', np.arange(3, dtype='int'), (col, )).to_dataframe()
```
Comment (ajtritt) [inline tests/unit/pynwb_tests/test_core.py]: Can you add assert statement checks to make sure the dataframe is converted appropriately?

**Review by ajtritt — CHANGES_REQUESTED** [review]

File: src/pynwb/core.py (hunk 1 above)
Comment (ajtritt) [reply src/pynwb/core.py:1244]: `tolist` makes a list of lists.... I'm not sure if that's preferable to a list of numpy arrays. @bendichter ?

**Review by ajtritt — CHANGES_REQUESTED** [review]

**Comment by bendichter** [comment]
Yeah I prefer a list of arrays

File: src/pynwb/core.py (hunk 1 above)
Comment (oruebel) [reply src/pynwb/core.py:1244]: Ok, I see. Thanks for the clarification, that makes sense.

**Review by oruebel — APPROVED** [review]

**Review by ajtritt — APPROVED** [review]

**Pull request merged**